master/internal/resourcemanagers/agent/agent.go

```diff
@@ -574,24 +575,27 @@ func (a *agent) handleContainersReattached(
 	recovered := map[cproto.ID]aproto.ContainerReattachAck{}
 
 	for _, containerRestored := range agentStarted.ContainersReattached {
-		if containerRestored.Failure == nil {
-			cid := containerRestored.Container.ID
-			_, ok := a.agentState.containerAllocation[cid]
-
-			if ok {
-				if a.agentState.containerState[cid].State == containerRestored.Container.State {
-					recovered[cid] = containerRestored
-				} else {
-					ctx.Log().Warnf(
-						"reattached container %s has changed state: %s to %s",
-						cid, a.agentState.containerState[cid].State,
-						containerRestored.Container.State)
-				}
-			}
-		} else {
+		if containerRestored.Failure != nil {
 			ctx.Log().Infof(
 				"agent failed to restore container: %v: %v",
 				containerRestored.Container.ID, containerRestored.Failure.ErrMsg)
+			continue
+		}
+
+		cid := containerRestored.Container.ID
+		_, ok := a.agentState.containerAllocation[cid]
+
+		if !ok {
+			continue
+		}
+
+		if a.agentState.containerState[cid].State == containerRestored.Container.State {
+			recovered[cid] = containerRestored
+		} else {
+			ctx.Log().Warnf(
+				"reattached container %s has changed state: %s to %s",
+				cid, a.agentState.containerState[cid].State,
+				containerRestored.Container.State)
 		}
```

Comment:
I think an advantage of doing this religiously is that it makes future diffs more likely to be additive, and that usually makes the change easier, too.
```suggestion
		cid := containerRestored.Container.ID
		_, ok := a.agentState.containerAllocation[cid]
		if !ok {
			continue
		}

		if a.agentState.containerState[cid].State != containerRestored.Container.State {
			ctx.Log().Warnf(
				"reattached container %s has changed state: %s to %s",
				cid, a.agentState.containerState[cid].State,
				containerRestored.Container.State)
			continue
		}

		recovered[cid] = containerRestored
```

Reply:
this is mostly a personal opinion, though, I know that.